week0002/pertsodian/java/src/SpiralMatrixGenerator.java

```diff
@@ -0,0 +1,39 @@
+public class SpiralMatrixGenerator {
+  public int[][] generate(int rows, int cols) {
+    if (rows == 0 || cols == 0)
+      throw new IllegalArgumentException("Matrix rols/cols must be positive");
+
+    int[][] matrix = new int[rows][cols];
+
+    int layer = 0;
+    int startElement = 1;
+    while (layer <= rows/2 && layer <= cols/2)
+      startElement = fillLayer(matrix, layer++, startElement);
+
+    return matrix;
+  }
+
+  private int fillLayer(int[][] matrix, int layer, int startElement) {
+    assert matrix.length > 0;
+    assert matrix[0].length > 0;
+
+    int rows = matrix.length;
+    int cols = matrix[0].length;
+
+    for (int i=layer; i<=cols-1-layer; i++)
```

Comment:
I personally prefer to read:
`for (int i = layer; i <= cols-1-layer; i++)`

`for (int i = layer; i <= cols - 1 - layer; i++)`
would be too much spacing in my opinion